ibazel/command/default_command.go

```diff
@@ -43,19 +46,26 @@ func DefaultCommand(startupArgs []string, bazelArgs []string, target string, arg
 }
 
 func (c *defaultCommand) Terminate() {
-	if c.pg != nil && !subprocessRunning(c.pg.RootProcess()) {
+	if !c.IsSubprocessRunning() {
+		c.pg = nil
 		return
 	}
-
-	// Kill it with fire by sending SIGKILL to the process PID which should
-	// propagate down to any subprocesses in the PGID (Process Group ID). To
-	// send to the PGID, send the signal to the negative of the process PID.
-	// Normally I would do this by calling c.cmd.Process.Signal, but that
-	// only goes to the PID not the PGID.
-	c.pg.Kill()
-	c.pg.Wait()
-	c.pg.Close()
+	if c.terminating {
```

Comment:
Have you considered using a [`sync.Once`](https://golang.org/pkg/sync/#Once) to perform this work instead of tracking the state in a bool? It is designed for exactly this purpose.

Reply:
`sync.Once` seems like better approach. 👍